### PR TITLE
show vocabularies release version

### LINKED ISSUE #13: Version on Download
On the ATHENA download screen, could you please put the version of the Vocabulary so the user knows before they download the whole thing?

### PR DESCRIPTION
**Pull request by acumarav**
show vocabularies release version in UI
Closes #13

### CODE REVIEW

File: src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java [hunk 1]
```diff
@@ -91,38 +93,32 @@
...
+        this.userService = userService;
+        this.vocabularyConversionService = vocabularyConversionService;
+        this.vocabularyService = vocabularyService;
     }
 
+
     @ApiOperation("Get vocabularies.")
     @RequestMapping(produces = MediaType.APPLICATION_JSON_VALUE, method = RequestMethod.GET)
-    public ResponseEntity<List<UserVocabularyDTO>> getAllForCurrentUser() throws Exception {
+    public ResponseEntity<List<UserVocabularyDTO>> getAllForCurrentUser() {
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java]: Well, while code has refactored why not to simply return entity w/o redundant wrappers such as `ResponseEntity`?

File: src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java [hunk 2]
```diff
@@ -174,20 +172,20 @@ public void getAllFiles(
             throws PermissionDeniedException {
 
         final AthenaUser user = userService.getUser(principal);
-        return new ResponseEntity<>(vocabularyService.getDownloadHistory(user.getId()), OK);
+        return ResponseEntity.ok(vocabularyService.getDownloadHistory(user.getId()));
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java]: entity

File: src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java [hunk 3]
```diff
@@ -209,7 +207,7 @@ public ResponseEntity checkBundle(@PathVariable("id") Long bundleId)
         AthenaUser currentUser = userService.getCurrentUser();
         vocabularyService.checkBundleUser(currentUser, bundle);
         vocabularyService.checkBundleVocabularies(bundle, currentUser.getId());
-        return new ResponseEntity<>(new LicenseExceptionDTO(true), HttpStatus.OK);
+        return ResponseEntity.ok(new LicenseExceptionDTO(true));
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java]: entity

File: src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java [hunk 4]
```diff
@@ -225,7 +223,7 @@ public ResponseEntity checkBundle(@PathVariable("id") Long bundleId)
         List<UserLicensesDTO> dtos = converterUtils.convertList(users.getContent(), UserLicensesDTO.class);
         CustomPageImpl<UserLicensesDTO> resultPage = new CustomPageImpl<>(dtos, pageRequest, users.getTotalElements());
 
-        return new ResponseEntity<>(resultPage, OK);
+        return ResponseEntity.ok(resultPage);
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java]: entity

File: src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java [hunk 5]
```diff
@@ -234,7 +232,7 @@ public ResponseEntity checkBundle(@PathVariable("id") Long bundleId)
     public ResponseEntity<List<VocabularyDTO>> suggestLicenses(@RequestParam("userId") Long userId) {
         //PENDING licenses are added -> do not need to suggest
         final List<VocabularyDTO> vocabularies = vocabularyConversionService.getUnavailableVocabularies(userId, true);
-        return new ResponseEntity<>(vocabularies, OK);
+        return ResponseEntity.ok(vocabularies);
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java]: entity

File: src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java [hunk 6]
```diff
@@ -320,6 +318,14 @@ public ResponseEntity notifyAboutUpdates(
         final AthenaUser user = userService.getUser(principal);
         List<Notification> notifications = vocabularyService.getNotifications(user.getId());
         List<VocabularyDTO> vocabularyDTOs = converterUtils.convertList(notifications, VocabularyDTO.class);
-        return new ResponseEntity<>(vocabularyDTOs, HttpStatus.OK);
+        return ResponseEntity.ok(vocabularyDTOs);
+    }
+
+    @GetMapping(value = "/releaseVersion")
+    public ResponseEntity<VocabularyVersionDTO> releaseVersion() {
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java]: entity

File: src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java [hunk 7]
```diff
@@ -320,6 +318,14 @@ public ResponseEntity notifyAboutUpdates(
...
+
+        String vocabularyVersion = vocabularyService.getVocabularyReleaseVersion();
+
+        return ResponseEntity.ok(new VocabularyVersionDTO(vocabularyVersion));
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java]: Pls, implement converter

File: src/main/java/com/odysseusinc/athena/model/common/AthenaConstants.java [hunk 8]
```diff
@@ -0,0 +1,9 @@
+package com.odysseusinc.athena.model.common;
+
+public class AthenaConstants {
+
+    private AthenaConstants() {
+    }
+
+    public static final String VOCABULARY_RELEASE_VERSION_ID = "None";
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/model/common/AthenaConstants.java]: Hmm, just not to misguide let's change the name of constant
It's not a vocabulary release id, it's OMOP Vocabulary ID

File: src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java [hunk 9]
```diff
@@ -169,6 +160,15 @@ public void checkBundleVocabularies(DownloadBundle bundle, Long userId) {
         checkBundleVocabularies(bundleVocabularyV4Ids, userId);
     }
 
+    @Override
+    public String getVocabularyReleaseVersion() {
+
+        final VocabularyV5 releaseVersion = vocabularyRepository.findOne(VOCABULARY_RELEASE_VERSION_ID);
+        LOGGER.debug("Current Vocabularies Release: {} {}: {}", releaseVersion.getId(), releaseVersion.getName(), releaseVersion.getVersion());
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java]: NPE here when no OMOP vocabulary in the database

File: src/main/java/com/odysseusinc/athena/service/mail/MailSender.java [hunk 10]
```diff
@@ -78,7 +78,8 @@ public void send(String email, Map<String, Object> parameters) throws MessagingE
             helper.setSubject(getSubject());
             helper.setFrom(from, notifier);
             helper.setTo(email);
-            helper.setText(contentBuilder.build(getTemplateName(), parameters), true);
+            final String emailBody = contentBuilder.build(getTemplateName(), parameters);
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/service/mail/MailSender.java]: What this fixes?

File: src/main/java/com/odysseusinc/athena/api/v1/controller/converter/vocabulary/StringToVocabularyVersionDTOConverter.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
+package com.odysseusinc.athena.api.v1.controller.converter.vocabulary;
+
+import com.odysseusinc.athena.api.v1.controller.dto.vocabulary.VocabularyVersionDTO;
+import org.springframework.beans.factory.InitializingBean;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.convert.converter.Converter;
+import org.springframework.core.convert.support.GenericConversionService;
+
+public class StringToVocabularyVersionDTOConverter implements Converter<String, VocabularyVersionDTO>, InitializingBean {
```
Comment (wivern) [inline src/main/java/com/odysseusinc/athena/api/v1/controller/converter/vocabulary/StringToVocabularyVersionDTOConverter.java]: Converter should appear in application context in any way,
e.g. with `@Component` annotation

File: src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java [hunk 12]
```diff
@@ -169,6 +161,19 @@ public void checkBundleVocabularies(DownloadBundle bundle, Long userId) {
         checkBundleVocabularies(bundleVocabularyV4Ids, userId);
     }
 
+    @Override
+    public String getOMOPVocabularyVersion() {
+
+        final VocabularyV5 omopVocabularyVersion = vocabularyRepository.findOne(OMOP_VOCABULARY_ID);
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/athena/service/impl/VocabularyServiceImpl.java]: This is vocabulary, not version. Why named as **Version** then?

File: src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java [hunk 13]
```diff
@@ -320,6 +315,14 @@ public ResponseEntity notifyAboutUpdates(
         final AthenaUser user = userService.getUser(principal);
         List<Notification> notifications = vocabularyService.getNotifications(user.getId());
         List<VocabularyDTO> vocabularyDTOs = converterUtils.convertList(notifications, VocabularyDTO.class);
-        return new ResponseEntity<>(vocabularyDTOs, HttpStatus.OK);
+        return ResponseEntity.ok(vocabularyDTOs);
+    }
+
+    @GetMapping(value = "/releaseVersion")
```
Comment (pavgra) [inline src/main/java/com/odysseusinc/athena/api/v1/controller/VocabularyController.java]: The whole application utilizes hyphens in URLs and you now introduce camel case. Why?

**Pull request merged**